src/socket/tcp/congestion_controller/cubic.rs

```diff
@@ -0,0 +1,312 @@
+use crate::time::Instant;
+
+use super::CongestionController;
+
+// Constants for the Cubic congestion control algorithm.
+// See RFC 8312.
+const BETA_CUBIC: f64 = 0.7;
+const C: f64 = 0.4;
```

Comment:
These `f64`s will be incredibly inefficient on most small embedded systems, being essentially unusable.

(By the way, it makes no sense to propose adding a generic parameter to reduce on call dispatch overhead while leavign in the massive f64 softfloat overhead. The latter will dwarf just about anything else...)

Reply:
A fixed point implementation would be ideal. Without it, the usefulness of the cubic congestion controller is quite limited, enough that it should be called out in the documentation as "do not use this on your STM32".

Reply:
I have inserted a warning message as follows to the document of `set_congestion_control()`.

```Rust
    /// `CongestionControl::Cubic` uses `f64` internally, which is less efficient on some platforms like STM32.
    pub fn set_congestion_control(&mut self, congestion_control: CongestionControl) {}
```

Reply:
I have added `socket-tcp-cubic` feature as discussed https://github.com/smoltcp-rs/smoltcp/pull/907#discussion_r1497091986 .
Cubic is disabled by default, and users cannot use it unless `socket-tcp-cubic` feature is enabled.
So, unintentional behavior can be avoided.